### ISSUE TITLE
Release the depth filter for better performance

### ISSUE THREAD

**Issue by tfoote** [orbbec/OpenNI2#2]
From this thread: https://3dclub.orbbec3d.com/t/astra-openni2-under-arm-platform-raspberry-pi-2-ubuntu-14-04/282/2 there's an unreleased filter that's not in the github sources. Without that the open source driver does not perform as well. 

This was first reported on the ROS review thread at http://discourse.ros.org/t/new-depth-sensor-orbbec-astra-call-for-testing/252/4

**Comment by mintar** [comment]
Any news on this? We cannot seriously consider the Astra as a viable replacement for our Xtions without this filter.

**Comment by facontidavide** [comment]
Hi, we still face the same problem: softfilter uses a large amount of CPU and we were not able to disable it following the instructions [here](https://github.com/orbbec/ros_astra_camera).

PLEASE either allow us to actually disable the filter or/and release the source code :)

**Comment by facontidavide** [comment]
Can you at LEAST give us a hint about what the filter is doing? is it a spatial filter or a temporal one?

**Comment by jlucier** [comment]
@facontidavide Did you end up resolving your performance issues?

I'm trying to use the AstraSDK (v2.1.1) directly rather than the [ROS node](https://github.com/orbbec/ros_astra_camera) and similarly observing wayyyy too much CPU usage. It seems like the SDK may also be doing some extra filtering on the depth which I can't afford, however I can't find anything documenting it or how I might be able to reduce the CPU usage.

**Comment by facontidavide** [comment]
Yes, I did solve it. Doing reverse engineering of the closed source algorithm and reimplementing it myself.

Unfortunately I can not share the code, because it was done at work, but my suggestion is that you disable their filter and you implement your own.

**Comment by facontidavide** [comment]
Let me add that I insisted for weeks with the technical support of Astra that if they release the code, I will share with them how to optimize it and benefit the entire community, and they refused.

**Comment by jlucier** [comment]
@facontidavide Excuse me if this is a dumb question, I'm not entirely sure how all these things play together.

Are you using OpenNI2 or the AstraSDK to communicate with your camera? I'm using the AstraSDK and cannot figure out how to disable filtering (or whether it's even happening at all).

**Comment by facontidavide** [comment]
You do care. The data without the filter is garbage.

You need to find the file orbbec.ini installed somewhere in your computer and modify to disable the filter
